src/CombFilter/CombFilter.cpp

```diff
@@ -0,0 +1,145 @@
+//
+//  CombFilter.cpp
+//  CombFilter
+//
+//  Created by Vedant Kalbag on 09/02/22.
+//
+#include "ErrorDef.h"
+#include "CombFilter.h"
+
+
+// =====================================================================================================
+// CCombFilterBase method definitions
+CCombFilterBase::CCombFilterBase(int delayLength, int iNumChannels, float gainValue)
+{
+    m_iNumChannels = iNumChannels;
+    m_fGainValue = gainValue;
+    m_iDelayValueSamples = delayLength;
+    m_RingBuffer = new CRingBuffer<float>* [m_iNumChannels];
+    for (int i=0;i<m_iNumChannels;i++)
+    {
+        m_RingBuffer[i] = new CRingBuffer<float>(m_iDelayValueSamples);
+        m_RingBuffer[i]->reset();
+    }
+}
+//CCombFilterBase::~CCombFilterBase()
+//{
+//}
+
+//Error_t CCombFilterBase::init(int iNumChannels, float delayValue, float gainValue)
+//{
+//
+//}
+
+Error_t CCombFilterBase::setGain(float fGainValue)
+{
+    //Set gainValue to fGainValue
+    m_fGainValue = fGainValue;
+    return Error_t::kNoError;
+}
+
+float CCombFilterBase::getGain()
+{
+    return m_fGainValue;
+}
+
+Error_t CCombFilterBase::setDelay(int iDelayValueSamples)
+{
+    //Set delayValue to iDelayValue
+    m_iDelayValueSamples = iDelayValueSamples;
+    return Error_t::kNoError;
+}
+
+int CCombFilterBase::getDelay()
+{
+    return m_iDelayValueSamples;
+}
+
+Error_t CCombFilterBase::reset()
+{
+    // reset values
+//    m_iDelayValueSamples = 0;
+//    m_fGainValue = 0.5f;
+
+}
+
+// =====================================================================================================
+// CCombIIRFilter definition
+class CCombIIRFilter : public CCombFilterBase
+{
+public:
+    CCombIIRFilter(int delayLength, int iNumChannels, float gain);
+    ~CCombIIRFilter();
+protected:
+//    int m_iNumChannels;
+    Error_t process (float **ppfInputBuffer, float **ppfOutputBuffer, int iNumberOfFrames) override;
+};
```

Comment:
I would prefer full class declaration in the header

Reply:
The idea of separating this out was so that the end user has no information about the class structure of the derived classes, since they are not interacting with it directly. Although they do use the interface class and not the base class either, so I guess this may be a bit redundant